stock_orderpoint_mto_as_mts/models/product_product.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_warehouse_missing_orderpoint_for_mto(self):
+        res = {}
+        wh_obj = self.env["stock.warehouse"]
+        for product in self:
+            wh_domain = [("mto_as_mts", "=", True)]
+            if product.company_id:
+                wh_domain.append(("company_id", "=", product.company_id.id))
+            wh = wh_obj.search(wh_domain)
+            wh_orderpoint = product.orderpoint_ids.partition("warehouse_id").keys()
+            wh_wo_orderpoint = wh.filtered(lambda r: r not in wh_orderpoint)
+            if wh_wo_orderpoint:
+                res[product] = wh_wo_orderpoint
+        return res
+
+    def _create_default_orderpoint_for_mto(self, warehouses):
+        self.ensure_one()
+        orderpoints = self.env["stock.warehouse.orderpoint"]
+        orderpoint_obj = self.env["stock.warehouse.orderpoint"]
+        for warehouse in warehouses:
+            orderpoint = orderpoint_obj.with_context(active_test=False).search(
+                [
+                    ("product_id", "=", self.id),
+                    (
+                        "location_id",
+                        "=",
+                        warehouse._get_locations_for_mto_orderpoints().id,
+                    ),
+                ],
+                limit=1,
```

Comment:
To prevent useless sorting cost, always better to set order when limit=1
```suggestion
                limit=1, order="id"
```